haskell-best-practices.md

```diff
@@ -396,6 +396,37 @@ f (ThingA a) = a == 0
 f (ThingB b) = if b then 1 else 2
 ```
 
+## Avoid `OverloadedLists` and `MonoTraversable`
+
+`OverloadedLists` work from a typelcass `IsList`:
+
+```
+class IsList l where
+type Item l
+fromList  :: [Item l] -> l
+toList    :: l -> [Item l]
+```
+  
+The `mono-traversable` package defines a series typeclasses:
+
+```
+type family Element mono
+
+class MonoFoldable mono where
+ofoldMap :: Monoid m => (Element mono -> m) -> mono -> m 
+...
+```
+  
+You can see that both of these typeclasses utilize type families. `IsList` has an associated type `Item` and `MonoFoldable` uses an open type family `Element`. They are nearly identical in their utilization of these techniques. I'll focus mainly on `mono-traversable`, but arguments against it are very transferable to `OverloadedLists`.
```

Comment:
I know this is from a slack comment, which is probably why it's somewhat colloquial. How about

> Each of these typeclasses rely on type families:
> * `IsList` has an associated type `Item` for representing the element of the list
> * `MonFoldable` uses an open type family `Elem` for representing the element of `mono`

You can probably lead into the next paragraph with

> Unfortunately, type families do not have great type inference. Furthermore, `mono-traversable` exports a number of replacements for common `Prelude` functions which rely on these type families.